docs/upgrades/v0.14-to-v1-upgrade/upgraded-examples/src/test/kotlin/examples/FunctionChanges.kt

```diff
@@ -0,0 +1,160 @@
+        assertEquals("HELLO", result.string)
+    }
+
+    @Test
+    fun `implementing aggregation functions`() {
+        // In PLK v1, aggregation functions extend Agg abstract class with Datum
+        class Count : Agg() {
+            // Check details in fun `creating function signatures`()
+            override fun getSignature(): RoutineSignature {
+                return RoutineSignature(
+                    "count",
+                    emptyList(),
+                    PType.integer()
+                )
+            }
+
+            // In PLK v1, accumulator is a standalone interface works with Datum arrays
+            override fun getAccumulator(): Accumulator {
+                return object : Accumulator {
+                    private var count = 0
+
+                    override fun next(args: Array<Datum>) {
+                        count++
+                    }
+
+                    override fun value(): Datum {
+                        return Datum.integer(count)
+                    }
+                }
+            }
+        }
+
+        val countFunc = Count()
+        val acc = countFunc.accumulator
+        acc.next(emptyArray())
+        acc.next(emptyArray())
+        assertEquals(2, acc.value().int)
+    }
+
+    @Test
+    fun `working with function overloads`() {
+        // In PLK v1, function overloading is handled by FnOverload and AggOverload classes
+        // The recommended approach is to use FnOverload.Builder multiple times for different types
+        // Create multiple overloads for the same function name with different parameter types
+        val absIntOverload = FnOverload.Builder("abs")
+            .addParameter(PType.integer())
+            .returns(PType.integer())
+            .body { args ->
+                val value = args[0].int
+                Datum.integer(abs(value))
+            }
+            .build()
+
+        val absDoubleOverload = FnOverload.Builder("abs")
+            .addParameter(PType.doublePrecision())
+            .returns(PType.doublePrecision())
+            .body { args ->
+                val value = args[0].double
+                Datum.doublePrecision(abs(value))
+            }
+            .build()
+
+        // Simulate catalog storage (in real implementation, this would be in your catalog class)
+        val functionOverloads = mapOf("abs" to listOf(absIntOverload, absDoubleOverload))
+        // Get all Abs overloads
+        val absOverloads = functionOverloads["abs"]!!
+        // Get instances for specific argument types
+        val absIntFn = absOverloads[0].getInstance(arrayOf(PType.integer()))
+        val absDoubleFn = absOverloads[1].getInstance(arrayOf(PType.doublePrecision()))
+        // Test different overloads with their respective types
+        val intResult = absIntFn!!.invoke(arrayOf(Datum.integer(-5)))
+        val doubleResult = absDoubleFn!!.invoke(arrayOf(Datum.doublePrecision(-3.14)))
+        assertEquals(5, intResult.int)
```

Comment:
Probably can also include an example call when an overload is not implemented (e.g. abs with a string argument)